odoo/cli/module.py

```diff
@@ -0,0 +1,163 @@
+            case 'force-demo':
+                self._force_demo(parsed_args)
+
+    def _check_modules_on_disk(self, parser, parsed_args):
+        """ Check modules on disk before starting the Db environment """
+        initialize_sys_path()
+        modules = set(parsed_args.modules)
+        valid_modules = {
+            module
+            for module in modules
+            if get_module_path(module, display_warning=False)
+        }
+        if invalid_modules := modules - valid_modules:
+            _logger.warning("Some of the modules weren't found on the filesystem: %s", ", ".join(invalid_modules))
+        if not valid_modules:
+            parser.error(f"No modules to {parsed_args.subcommand}")
+        return valid_modules
+
+    def _get_modules(self, env, domain, action):
+        """ Get modules from the db, depending on the state """
+        Module = env['ir.module.module']
+        Module.update_list()
+        return Module.search(domain)
+
+    def _install(self, parsed_args):
+        parsed_args.modules = self._check_modules_on_disk(self.install_parser, parsed_args)
+        with Registry.new(parsed_args.db_name).cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            if modules := self._get_modules(env, domain=[
+                ('state', 'in', ('uninstalled', 'to_install', 'to upgrade')),
+                ('name', 'in', parsed_args.modules),
+            ], action='be installed'):
+                modules.button_immediate_install()
+
+    def _upgrade(self, parsed_args):
+        parsed_args.modules = self._check_modules_on_disk(self.upgrade_parser, parsed_args)
+        with Registry.new(parsed_args.db_name).cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            domain = [('state', 'in', ('installed', 'to_install', 'to upgrade')), ('name', 'in', parsed_args.modules)]
+            if modules := self._get_modules(env, domain, action='be upgraded'):
+                modules.button_immediate_upgrade()
+
+    def _uninstall(self, parsed_args):
+        parsed_args.modules = self._check_modules_on_disk(self.uninstall_parser, parsed_args)
+        with Registry.new(parsed_args.db_name).cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            domain = [('state', 'in', ('installed', 'to upgrade', 'to remove')), ('name', 'in', parsed_args.modules)]
+            if modules := self._get_modules(env, domain, action='be uninstalled'):
+                modules.button_immediate_uninstall()
+
+    def _force_demo(self, parsed_args):
+        parsed_args.modules = self._check_modules_on_disk(self.force_demo_parser, parsed_args)
+        with Registry.new(parsed_args.db_name).cursor() as cr:
+            env = odoo.api.Environment(cr, odoo.SUPERUSER_ID, {})
+            domain = [('state', 'in', ('installed', 'to upgrade', 'to remove')), ('demo', '=', False)]
+            if parsed_args.modules:
```

Comment:
Do you want to support `--force-demo account`?
If so, it would be quite complicated. You need get `modules.upstream_dependencies()` and install demo data if their `demo` data are not installed.

I think it would be simpler and better to reuse the `force_demo` like you did before. And improve the logic in `force_demo`. And for developers when they use the CLI with --force-demo, intuitively they would expect the `force_demo` function to be called.

Reply:
I removed, and just called `force_demo`